src/Datadog.Trace.ClrProfiler.Managed/Extensions.cs

```diff
@@ -42,17 +42,17 @@ public static object Trace(this Span span, Func<object> func, Action<Exception>
                     var genericArgs = GetGenericTypeArguments(typeof(Task<>), typ);
                     if (genericArgs.Length == 1)
                     {
-                        if (!_traceTaskMethods.TryGetValue(genericArgs[0], out var traceTask))
+                        if (!_traceTaskMethods.TryGetValue(genericArgs[0], out var traceTaskAsync))
```

Comment:
It would be nice to have a unit test checking this scenario.

Reply:
Excellent advice. I added unit tests and found a few issues with `Task<T>` here and fixed them. Thanks!